components/org.wso2.is.password.reencrypt/src/main/java/secondaryUserStore/migrator/UserStorePasswordMigrator.java

```diff
@@ -0,0 +1,174 @@
+            String filename = listOfFiles[i].getName();
+            String filePath = listOfFiles[i].getPath();
+            if (filename.matches("\\d+")) {
+                filePaths.add(filePath);
+            }
+        }
+        return filePaths;
+    }
+
+    public void migrate() throws MigrationClientException {
+
+        log.info(Constant.MIGRATION_LOG + "Migration starting on Secondary User Stores");
+        updateSuperTenantConfigs();
+        updateTenantConfigs();
+    }
+
+    private void updateTenantConfigs() throws MigrationClientException {
+
+        try {
+            String carbonHome = System.getProperty(Constant.CARBON_HOME);
+            List<String> tenantsPaths = findSubfiles(Paths
+                    .get(carbonHome, new String[]{"repository", "tenants"}).toString());
+            for (String tenant : tenantsPaths) {
+
+                try {
+                    File[] userStoreConfigs = getUserStoreConfigFiles(tenant);
+                    for (File file : userStoreConfigs) {
+                        if (file.isFile()) {
+                            updatePassword(file.getAbsolutePath());
+                        }
+                    }
+                } catch (FileNotFoundException | CryptoException e) {
+                    String msg =
+                            "Error while updating secondary user store password for tenant: " + tenant.split("/")[-1];
+                    log.error(msg, e);
+                }
+            }
+        } catch (Exception e) {
+            throw new MigrationClientException("Error while getting tenants for migration", e);
+        }
+    }
+
+    private void updateSuperTenantConfigs() {
+
+        try {
+            String carbonHome = System.getProperty(Constant.CARBON_HOME);
+            String userStorePath = Paths.get(carbonHome, new String[]{"repository", "deployment", "server",
+                    "userstores"})
+                    .toString();
+            File[] userStoreConfigs = new File(userStorePath).listFiles();
+            if (userStoreConfigs == null) {
+                userStoreConfigs = new File[0];
+            }
+            for (File file : userStoreConfigs) {
+                if (file.isFile()) {
+                    updatePassword(file.getAbsolutePath());
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error while updating secondary user store password for super tenant", e);
+        }
+    }
+
+    private File[] getUserStoreConfigFiles(String parentPath) {
+
+        String userStorePath;
+        userStorePath = Paths
+                .get(parentPath, "userstores")
+                .toString();
+
+        File[] files = new File(userStorePath).listFiles();
+        return files != null ? files : new File[0];
+    }
+
+    private void updatePassword(String filePath) throws FileNotFoundException, CryptoException {
+
+        XMLStreamReader parser = null;
+        FileInputStream stream = null;
```

Comment:
FileInputStream is AutoCloseable, So can use try with resources

Reply:
PR send #6